### DISCUSSION TITLE
Issue related to strongswan running in network namespace

### DISCUSSION THREAD

**Discussion by vlrk** [strongswan/strongswan#1115] [Q&A]
Hi 

I want to run the strongswan in the namespace
(https://docs.strongswan.org/docs/5.9/howtos/nameSpaces.html)

In configure gave options as below
./configure --sysconfdir=/etc --with-piddir=/etc/run

namespace  I have is netns1

Now my etc folder looks like "/etc/netns/netns1/" and run folder is "/etc/netns/netns1/run"
All my configs are now moved to /etc/netns/netns1 .

In namespace when I try to start ipsec "ip netns exec netns1 ipsec start" , gives me error as below 

Bind /etc/netns/netns1/cups -> /etc/cups failed: No such file or directory
Bind /etc/netns/netns1/ipsec.d -> /etc/ipsec.d failed: No such file or directory
Bind /etc/netns/netns1/strongswan.conf -> /etc/strongswan.conf failed: No such file or directory
Bind /etc/netns/netns1/strongswan.d -> /etc/strongswan.d failed: No such file or directory
Bind /etc/netns/netns1/swanctl -> /etc/swanctl failed: No such file or directory
no files found matching '/etc/strongswan.conf'
abort initialization due to invalid configuration
Starting strongSwan 5.7.2 IPsec [starter]...
expanding file pattern '/etc/ipsec.d/*.conf' failed: No such file or directory
# deprecated keyword 'plutodebug' in config setup
# deprecated keyword 'virtual_private' in config setup
### 2 parsing errors (0 fatal) ###

It seems ipsec facing issue while trying to use the namespace specific folder 

Any inputs on this would be really helpfull.

Thanks in advance.
vlrk

**Comment by vlrk** [comment]
I got this working , by keeping original set of files in the /etc .

But when I try to change 

"./configure --sysconfdir=/etc --with-piddir=/etc/run" to "./configure --sysconfdir=/var/etc --with-piddir=/var/etc/run" 

It does not work .

It takes only one instance and that too from /var/etc/run/charon.pid only

Any inputs here , how can I put my own conf and piddir here ?.

Thanks

**Comment by tobiasbrunner** [comment]
As documented (in [our docs](https://docs.strongswan.org/docs/5.9/howtos/nameSpaces.html) and the man page for `ip netns`), only files in `/etc` are mapped automatically. For other files/directories you have to do your own bind mounts (not sure if that works with `ip netns`, you might have to use `nsenter` manually).
